doc/source/exclude_patterns.md

```diff
@@ -0,0 +1,93 @@
+# Exclude Pattern Settings 
+- Optional argument for [in-toto-run.](https://in-toto.readthedocs.io/en/latest/command-line-tools/in-toto-run.html) Path patterns to match paths that should not be recorded as 'materials' or 'products'.  Passed patterns override patterns defined in environment variables or config files (see ```ARTIFACT_EXCLUDE_PATTERNS``` documentation for additional info) 
+- Uses [PathSpec](http://python-path-specification.readthedocs.io) to compile [gitignore-style patterns](https://git-scm.com/docs/gitignore), implementing GitWildMatchPattern class which itself is derived from [Rsync's wildmatch](https://github.com/AndyA/rsync/blob/master/lib/wildmatch.c)   
+- If a directory is excluded, all its files and subdirectories are also excluded
+- Exclude patterns are likely to become command line arguments or part of a config file.
+- Pattern Format
+  - Include backslash (' \\ ') in front of first hash for patterns that begin with a literal hash
+  - Trailing spaces are ignored unless quote with backslash (' \\ ')
+  - Prefix '!' negates the pattern. Excluded patterns will become included again 
```

Comment:
> Excluded patterns will become included again

I think this can be a little confusing. Let's talk about it in terms of the files matched by the pattern as a whole.

Reply:
Reworded it as "Prefix '!' negates the pattern. Previously matched patterns will be unmatched". Let me know what you think!